guide/src/annotations.md

```diff
@@ -0,0 +1,113 @@
+# Annotations
+
+Duvet scans source code for special comments containing references to specification text. By default, the comment style is the following:
+
+```rust
+//= https://www.rfc-editor.org/rfc/rfc2324#section-2.1.1
+//# A coffee pot server MUST accept both the BREW and POST method
+//# equivalently.
+```
+
+If the default comment style is not compatible with the language being used, it can be changed in the [configuration](./config.md) with the `comment-style` field.
+
+The default type of annotation is `implementation`, meaning the reference is implementing the cited text. The type of annotation can be changed with the `type` parameter. Duvet supports the following annotation types:
+
+## `implementation`
+
+The source code is aiming to implement the cited text from the specification. This is the default annotation type.
+
+## `test`
+
+The source code is aiming to test that the program implements the cited text correctly.
+
+```rust
+//= https://www.rfc-editor.org/rfc/rfc2324#section-2.1.1
+//= type=test
+//# A coffee pot server MUST accept both the BREW and POST method
+//# equivalently.
+#[test]
+fn my_test() {
+    // TODO
+}
+```
+
+## `implication`
+
+The source code is both implementing and testing the cited text. This can be useful for requirements that are correct by construction. For example, let's say our specification says the following:
+
+```
+# Section
+
+The function MUST return a 64-bit integer.
+```
+
+In a strongly-typed language, this requirement is being both implemented and tested by the compiler.
+
+```rust
+//= my-spec.md#section
+//# The function MUST return a 64-bit integer.
+fn the_function() -> u64 {
+    42
+}
+```
+
+## `exception`
+
+The source code has defined an exception for a requirement and is explicitly choosing not to implement it. This could be for various reasons. For example, let's consider the following specification:
+
+```
+# Section
+
+Implementations MAY panic on invalid arguments.
+```
+
+In our example here, we've chosen _not_ to panic, but instead return an error. Annotations with the `exception` type are expected to provide a reason as to why the requirement is not being implemented.
```

Comment:
`Annotations with the `exception` type are expected`

I forget if "reason" is required for this, maybe make it clear one way or the other if its optional or required